File: federlang/comparison/go/prime10001st.go
package  main

import (
	"fmt"
	"math"
)

func IsPrime (primes []uint32, num uint32) bool {
	max_num := uint32(math.Sqrt (float64(num)))
	len_primes := len (primes)
	for  i := 0; i <  len_primes; i++ {
		if primes[i] > max_num {
			return true
		}


		if (num % primes[i]) == 0 {
			return false
		}
	}

	return true
}

func main() {
	var primes []uint32

	primes = append(primes, uint32(2))
	primes = append(primes, uint32(3))
	primes = append(primes, uint32(5))

	for i := uint32(6); len(primes) < 10001; i++ {
		if IsPrime(primes, i) {
			primes = append(primes, i)
		}
	}

	fmt.Printf("10001st prime: %d\n", primes[len(primes) -1])
}

File: federlang/comparison/python/prime10001st.py
from math import sqrt

def isPrime (primes, num):
    max_num = int (sqrt (num))
    
    for prime in primes:
        if max_num < prime:
            return True
         
        if num % prime == 0:
            return False
    
    print (num)
    return True

primes = [2, 3, 5]
current_num = 6
while len (primes) < 10001:
    if isPrime (primes, current_num):
        primes.append (current_num)
    
    current_num += 1

print ("10001st prime:", (current_num-1))
